plugins/modules/files/json_patch.py

```diff
@@ -0,0 +1,566 @@
+            - The path to the destination JSON file
+        required: False
+        type: str
+    operations:
+        description:
+            - A list of operations to perform on the JSON document
+        required: True
+        type: list
+    backup:
+        description:
+            - Copy the targeted file to a backup prior to patch
+        required: False
+        type: bool
+    unsafe_writes:
+        description:
+            - Allow Ansible to fall back to unsafe methods of writing files (some systems do not support atomic operations)
+        required: False
+        type: bool
+    pretty:
+        description:
+            - Write pretty-print JSON when file is changed
+        required: False
+        type: bool
+    create:
+        description:
+            - Create a file if it does not already exist
+        required: False
+        type: bool
+    create_type:
+        description:
+            - Initialize a newly created JSON file as an object or array
+        required: False
+        choices:
+            - "object"
+            - "array"
+        default: "object"
+        type: str
+'''
+
+
+EXAMPLES = r'''
+# These examples are using the following JSON:
+#
+# [
+#   {
+#     "foo": {
+#       "one": 1,
+#       "two": 2,
+#       "three": 3
+#     },
+#     "enabled": true
+#   },
+#   {
+#     "bar": {
+#       "one": 1,
+#       "two": 2,
+#       "three": 3
+#     },
+#     "enabled": false
+#   },
+#   {
+#     "baz": [
+#       {
+#         "foo": "apples",
+#         "bar": "oranges"
+#       },
+#       {
+#         "foo": "grapes",
+#         "bar": "oranges"
+#       },
+#       {
+#         "foo": "bananas",
+#         "bar": "potatoes"
+#       }
+#     ],
+#     "enabled": true
+#   }
+# ]
+#
+- name: add a fourth element to the "foo" object
+  json_patch:
+    src: "test.json"
+    dest: "test2.json"
+    operations:
+      - op: add
+        path: "/0/foo/four"
+        value: 4
+
+- name: remove the first object in the "baz" list of fruits
+  json_patch:
+    src: "test.json"
+    pretty: yes
+    operations:
+      - op: remove
+        path: "/2/baz/0"
+
+- name: move the "potatoes" value from the "baz" list to the "foo" object
+  json_patch:
+    src: "test.json"
+    backup: yes
+    operations:
+      - op: move
+        from: "/2/baz/2/bar"
+        path: "/0/foo/bar"
+
+- name: test that the "foo" object has three members
+  json_patch:
+    src: "test.json"
+    operations:
+      - op: test
+        path: "/0/foo/one"
+        value: 1
+      - op: test
+        path: "/0/foo/two"
+        value: 2
+      - op: test
+        path: "/0/foo/three"
+        value: 3
+'''
+
+
+RETURN = r'''
+changed:
+    description: whether the JSON object was modified
+    returned: always
+    type: bool
+tested:
+    description: the result of any included test operation
+    returned: when test operation exists
+    type: bool
+backup:
+    description: the name of the backed up file
```

Comment:
```suggestion
    description: The name of the backed up file
```
the same for the other descriptions in this block